dandiapi/api/tests/test_tasks.py

```diff
@@ -250,3 +252,108 @@ def test_validate_version_metadata_malformed_license(version: Version, asset: As
     assert version.validation_errors == [
         {'field': 'license', 'message': "'foo' is not of type 'array'"}
     ]
+
+
+@pytest.mark.django_db
+def test_unembargo_dandiset(
+    dandiset_factory,
+    draft_version_factory,
+    asset_factory,
+    embargoed_asset_blob_factory,
+    storage,
+    embargoed_storage,
+):
+    # Since both fixtures are parametrized, only proceed when they use the same storage backend
+    if type(storage) != type(embargoed_storage):
+        return
+
+    # Pretend like AssetBlob/EmbargoedAssetBlob were defined with the given storage
+    AssetBlob.blob.field.storage = storage
+    EmbargoedAssetBlob.blob.field.storage = embargoed_storage
+
+    # Create dandiset and version
+    dandiset: Dandiset = dandiset_factory(embargo_status=Dandiset.EmbargoStatus.EMBARGOED)
+    draft_version_factory(dandiset=dandiset)
+
+    # Create embargoed assets
+    embargoed_asset_blob: EmbargoedAssetBlob = embargoed_asset_blob_factory()
+    embargoed_asset: Asset = asset_factory(embargoed_blob=embargoed_asset_blob, blob=None)
+
+    # Assert properties before unembargo
+    assert embargoed_asset.embargoed_blob is not None
+    assert embargoed_asset.blob is None
+    assert embargoed_asset.embargoed_blob.etag != ''
+
+    # Run unembargo
+    draft_version = dandiset.draft_version
+    draft_version.assets.add(embargoed_asset)
+    tasks.unembargo_dandiset(dandiset.pk)
+    dandiset.refresh_from_db()
+    draft_version.refresh_from_db()
+
+    # Assert correct changes took place
+    assert dandiset.embargo_status == Dandiset.EmbargoStatus.OPEN
+    assert draft_version.metadata['access'] == 'dandi:OpenAccess'
+
+    asset: Asset = draft_version.assets.first()
+    assert asset.embargoed_blob is None
+    assert asset.blob is not None
+    assert asset.blob.etag == embargoed_asset_blob.etag
+
```

Comment:
Can you add an assertion that the asset.blob.name or whatever is the expected value? It should be something like:
```python
assert asset.blob.name == f'test-prefix/blobs/{asset.asset_id[:3]}/{asset.asset_id[3:6]}/{asset.asset_id}'
```

Reply:
582272e